lib/site_prism/element_container.rb

```diff
@@ -178,41 +184,44 @@ def create_waiter(element_name, *find_args)
       def create_nonexistence_waiter(element_name, *find_args)
         method_name = "wait_for_no_#{element_name}"
         create_helper_method(method_name, *find_args) do
-          define_method(method_name) do |timeout = Capybara.default_max_wait_time, *runtime_args|
-            result = Capybara.using_wait_time(timeout) do
+          define_method(method_name) do |timeout = max_wait_time, *runtime_args|
+            res = Capybara.using_wait_time(timeout) do
               element_does_not_exist?(*merge_args(find_args, runtime_args))
             end
-            raise_wait_for_no_if_failed(self, element_name.to_s, timeout, !result)
-            result
+            raise_wait_for_no_if_failed(self, element_name.to_s, timeout, !res)
+            res
           end
         end
       end
 
       def create_visibility_waiter(element_name, *find_args)
         method_name = "wait_until_#{element_name}_visible"
         create_helper_method(method_name, *find_args) do
-          define_method(method_name) do |timeout = Capybara.default_max_wait_time, *runtime_args|
-            unless element_exists?(*merge_args(find_args, runtime_args, visible: true, wait: timeout))
-              raise SitePrism::TimeOutWaitingForElementVisibility
-            end
+          define_method(method_name) do |timeout = max_wait_time, *runtime_args|
+            visibility_args = { visible: true, wait: timeout }
+            args = merge_args(find_args, runtime_args, **visibility_args)
+            next if element_exists?(*args)
```

Comment:
should this be `return` - I know how `next` works but we're not quite in an enumerable.? ignore if not relevant

Same for item below on line215

Reply:
nice catch, probably `return` will be better here since this is a method definition.
I was confused because `return` is used only in methods. But we are defining method here.